mirgecom/multiphysics/thermally_coupled_fluid_wall.py

```diff
@@ -0,0 +1,1041 @@
+from mirgecom.fluid import make_conserved
+from mirgecom.flux import num_flux_central
+from mirgecom.inviscid import inviscid_facial_flux_rusanov
+from mirgecom.viscous import viscous_facial_flux_harmonic
+from mirgecom.gas_model import (
+    make_fluid_state,
+    make_operator_fluid_states,
+)
+from mirgecom.navierstokes import (
+    grad_t_operator as fluid_grad_t_operator,
+    ns_operator,
+)
+from mirgecom.diffusion import (
+    DiffusionBoundary,
+    grad_operator as wall_grad_t_operator,
+    diffusion_operator,
+)
+
+
+class _TemperatureInterVolTag:
+    pass
+
+
+class _KappaInterVolTag:
+    pass
+
+
+class _GradTemperatureInterVolTag:
+    pass
+
+
+class _FluidOpStatesTag:
+    pass
+
+
+class _FluidGradTag:
+    pass
+
+
+class _FluidOperatorTag:
+    pass
+
+
+class _WallGradTag:
+    pass
+
+
+class _WallOperatorTag:
+    pass
+
+
+# FIXME: Import from mirgecom.boundary when
+# https://github.com/illinois-ceesd/mirgecom/pull/678 gets merged
+def _get_normal_axes(seed_vector):
+    from arraycontext import get_container_context_recursively
+    actx = get_container_context_recursively(seed_vector)
+    vec_dim, = seed_vector.shape
+
+    vec_mag = actx.np.sqrt(np.dot(seed_vector, seed_vector))
+    seed_vector = seed_vector / vec_mag
+
+    if vec_dim == 1:
+        return seed_vector,
+
+    if vec_dim == 2:
+        vector_2 = 0*seed_vector
+        vector_2[0] = -1.*seed_vector[1]
+        vector_2[1] = 1.*seed_vector[0]
+        return seed_vector, vector_2
+
+    if vec_dim == 3:
+        x_comp = seed_vector[0]
+        y_comp = seed_vector[1]
+        z_comp = seed_vector[2]
+        zsign = z_comp / actx.np.abs(z_comp)
+
+        a = vec_mag * zsign
+        b = z_comp + a
+
+        vector_2 = 0*seed_vector
+        vector_2[0] = a*b - x_comp*x_comp
+        vector_2[1] = -x_comp*y_comp
+        vector_2[2] = -x_comp*b
+        vec_mag2 = actx.np.sqrt(np.dot(vector_2, vector_2))
+        vector_2 = vector_2 / vec_mag2
+        x_comp_2 = vector_2[0]
+        y_comp_2 = vector_2[1]
+        z_comp_2 = vector_2[2]
+
+        vector_3 = 0*vector_2
+        vector_3[0] = y_comp*z_comp_2 - y_comp_2*z_comp
+        vector_3[1] = x_comp_2*z_comp - x_comp*z_comp_2
+        vector_3[2] = x_comp*y_comp_2 - y_comp*x_comp_2
+
+    return seed_vector, vector_2, vector_3
+
+
+# FIXME: Import from mirgecom.boundary when
+# https://github.com/illinois-ceesd/mirgecom/pull/678 gets merged
+def _get_rotation_matrix(principal_direction):
+    principal_axes = _get_normal_axes(principal_direction)
+    dim, = principal_direction.shape
+    comps = []
+
+    for d in range(dim):
+        axis = principal_axes[d]
+        for i in range(dim):
+            comps.append(axis[i])
+
+    from pytools.obj_array import make_obj_array
+    comps = make_obj_array(comps)
+    return comps.reshape(dim, dim)
+
+
+class InterfaceFluidSlipBoundary(PrescribedFluidBoundary):
+    """Interface boundary condition for the fluid side."""
```

Comment:
Looking through this, it could possibly directly inherit from the AdiabaticSlip to pick up the fluid boundary stuff.   If not then maybe at least MengaldoBoundaryCondition.   I hope we can avoid using PrescribedFluidBoundary.